### PR TITLE
add include/exclude tables option when creating graph

### PR DESCRIPTION
**Pull request by aja114**
I have only added a few tests for this new feature atm. I wanted to check in first if this is what you had in mind for this feature.

### CODE REVIEW

File: paracelsus/graph.py [hunk 1]
```diff
@@ -48,5 +52,62 @@ def get_graph_string(
...
+    include_tables: Set[str],
+):
+    """Check that all foreign keys in the included_tables set point to included tables."""
+    for table_name in include_tables:
+        for fk in metadata.tables[table_name].foreign_keys:
+            if fk.column.table.name not in include_tables:
+                raise RuntimeError(
+                    f"Table '{table_name}' contains a foreign key to '{fk.column.table.name}' "
+                    "which is excluded / not-included."
+                )
```
Comment (tedivm) [inline paracelsus/graph.py:101]: If the table has a foreign key to an excluded table I think we should still create a graph (and include the foreign key as a column of that graph), we should just not include the connection to the other table.

For instance, if we use the example graph from the readme but excluded the`users` table it should generate this graph:

```mermaid
erDiagram
 posts {
  CHAR(32) id PK
  CHAR(32) author FK
  TEXT content "nullable"
  DATETIME created
  BOOLEAN live "nullable"
 }

 comments {
  CHAR(32) id PK
  CHAR(32) author FK
  CHAR(32) post FK "nullable"
  TEXT content "nullable"
  DATETIME created
  BOOLEAN live "nullable"
 }

 posts ||--o{ comments : post
```

File: tests/test_graph.py [hunk 2]
```diff
@@ -1,8 +1,37 @@
+import pytest
 from paracelsus.graph import get_graph_string
 
 from .utils import mermaid_assert
 
 
 def test_get_graph_string(package_path):
-    graph_string = get_graph_string("example.base:Base", ["example.models"], [package_path], "mermaid")
+    graph_string = get_graph_string("example.base:Base", ["example.models"], set(), set(), [package_path], "mermaid")
```
Comment (tedivm) [inline tests/test_graph.py:8]: Since this is starting to get really long I'd prefer if we switched to named arguments rather than positional arguements. ruff/black will then format them so one arg is on each line.

File: paracelsus/graph.py [hunk 3]
```diff
@@ -48,5 +52,62 @@ def get_graph_string(
...
+            if not include_tables.issubset(all_tables):
+                non_existent_tables = include_tables - all_tables
+                raise ValueError(
+                    f"Some tables to include ({non_existent_tables}) don't exist within the found tables ({all_tables})."
+                )
+            return include_tables, all_tables - include_tables
+        case _:
+            raise ValueError(
+                f"Only one or none of include_tables ({include_tables}) or exclude_tables ({exclude_tables}) can contain values."
+            )
```
Comment (tedivm) [inline paracelsus/graph.py:87]: I'm finding the logic hard to follow here, and it's not really clear what the two return arguments are. Some additional comments would really help here.

**Review by tedivm — CHANGES_REQUESTED** [review]
Overall this is looking really good! I added some inline feedback, mostly a request for some more comments but I did make a request that we don't return an error if a table refers to an excluded table but instead just drop the connection.

Thanks for working on this!

File: paracelsus/graph.py (hunk 3 above)
Comment (aja114) [reply paracelsus/graph.py:87]: I added a more complete docstring, does this make it clearer?

File: tests/test_graph.py (hunk 2 above)
Comment (aja114) [reply tests/test_graph.py:8]: Done

File: paracelsus/graph.py (hunk 1 above)
Comment (aja114) [reply paracelsus/graph.py:101]: I was planning on not touching on the `transformers` package but this change requires it (not sure if there's another way to do this). So I removed the error and I'm skipping the connection when building the representations directly (if the project grows maybe having a common representation and then passing them to the transformer classes could be useful).

I think this also makes sense when a referenced model (table) was not imported and registered in `metadata`, I guess currently an unhandled error is raised inside the transformers.

**Comment by tedivm** [comment]
This all looks really good- was there anything else you wanted to do before switching off of a draft PR?

**Review by tedivm — APPROVED** [review]

**Comment by aja114** [comment]
I added two extra tests for the cli and some reference in the README. That's all I wanted to add

**Review by tedivm — APPROVED** [review]

**Comment by tedivm** [comment]
Thanks for the PR!
